### PR TITLE
cron lambda

### PR DESCRIPTION
**Pull request by saurabhjindal-SF**
- cdktf cron lambda
- bug fixes in previous stacks

### CODE REVIEW

**Comment by mayank0202** [comment]
@saurabhjindal-SF  Did you ran this stack on aws to check if everything is working? There is a .gen file which doesn't works in my system and i am not able to test it so thats why i am asking this...

File: aws-lambdas/iac/cdk/common/stacks/cron.stack.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import * as aws from "@cdktf/provider-aws";
+import { TerraformStack } from "cdktf";
+import { Construct } from "constructs";
+import * as random from "../../.gen/providers/random";
```
Comment (mayank0202) [inline aws-lambdas/iac/cdk/common/stacks/cron.stack.ts:4]: @jamescrowley321  @saurabhjindal-SF Do we import random like this? I am not a good guy on this so just a question

**Review by mayank0202 — APPROVED** [review]

File: aws-lambdas/iac/cdk/common/constructs/lambda.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        path: config.layerPath,
+        type: AssetType.ARCHIVE, // if left empty it infers directory and file
+      });
+      // Create Lambda Layer for function
+      const lambdaLayers = new aws.lambdaLayerVersion.LambdaLayerVersion(
+        this,
+        "lambda-layer",
+        {
+          filename: layerAsset.path,
+          layerName: `${id}-layers-${config.name}`,
```
Comment (jamescrowley321) [inline aws-lambdas/iac/cdk/common/constructs/lambda.ts:38]: Please make sure this matching our standard naming convention pattern.

File: aws-lambdas/iac/cdk/common/constructs/lambda.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        }
+      );
+
+      layers.push(lambdaLayers.arn);
+    }
+
+    // Create Lambda role
+    const role = new aws.iamRole.IamRole(this, "lambda-exec", {
+      name: `lambda-role-${id}-${config.name}`,
```
Comment (jamescrowley321) [inline aws-lambdas/iac/cdk/common/constructs/lambda.ts:47]: Please make sure this matching our standard naming convention pattern.

File: aws-lambdas/iac/cdk/common/constructs/lambda.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        role: role.name,
+      }
+    );
+
+    // Create Lambda function
+    const lambdaFunc = new aws.lambdaFunction.LambdaFunction(
+      this,
+      "lambda-function",
+      {
+        functionName: `cdktf-${id}-${config.name}`,
```
Comment (jamescrowley321) [inline aws-lambdas/iac/cdk/common/constructs/lambda.ts:70]: Please make sure this matching our standard naming convention pattern.

**Review by jamescrowley321 — CHANGES_REQUESTED** [review]

**Review by jamescrowley321 — APPROVED** [review]

**Pull request merged**